www/js/metrics.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+angular.module('emission.main.metrics',['nvd3', 'emission.services'])
+
+.controller('MetricsCtrl', function($scope, $ionicActionSheet, CommHelper) {
+    $scope.options = {
+        chart: {
+            type: 'multiBarChart',
+            height: 450,
+            margin : {
+                top: 20,
+                right: 20,
+                bottom: 40,
+                left: 55
+            },
+            showValues: true,
+            stacked: false,
+            x: function(d){ return d[0]; },
+            y: function(d){ return d[1]; },
+            /*
+            average: function(d) { 
+                var vals = d.values.map(function(item){
+                    return item[1];
+                });
+                return d3.mean(vals);
+            },
+            */
+
+            color: d3.scale.category10().range(),
+            // duration: 300,
+            useInteractiveGuideline: true,
+            // clipVoronoi: false,
+
+            xAxis: {
+                axisLabel: 'Date',
+                tickFormat: function(d) {
+                    return d3.time.format('%y-%m-%d')(new Date(d * 1000))
+                },
+                showMaxMin: false,
+                staggerLabels: true
+            },
+            yAxis: {
+              axisLabel: "Number",
+              axisLabelDistance: -10
+            },
+        }
+    };
+
+    $scope.data = [];
+
+    $scope.getMetrics = function() {
+      var data = {
+        freq: 'DAILY',
+        start_time: $scope.selectCtrl.fromDate,
+        end_time: $scope.selectCtrl.toDate,
+        metric: $scope.selectCtrl.metric
+      };
+      console.log("Sending data "+JSON.stringify(data));
+      CommHelper.getMetrics("local_date", data, function(response) {
+        if (angular.isDefined(response.user_metrics)) {
+          console.log("Got aggregate result "+response.user_metrics.length);
+          $scope.$apply(function() {
+              $scope.showCharts(response.user_metrics)
+          });
+        } else {
+          console.log("did not find aggregate result in response data "+JSON.stringify(response));
+        }
+      }, function(error) {
+        console.log("Got error %s while trying to read metric data" +
+        JSON.stringify(error));
+      });
+    };
+
+    $scope.showCharts = function(agg_metrics) {
+        var mode_bins = {};
+        agg_metrics.forEach(function(metric) {
+            for (var field in metric) {
+                // TODO: Consider creating a prefix such as M_ to signal
+                // modes. Is that really less fragile than caps, though?
+                // Here, we check if the string is all upper case by 
+                // converting it to upper case and seeing if it is changed
+                if (field == field.toUpperCase()) {
```

Comment:
Why do we need to check if the field is upper case? What happens if it is not uppercase?